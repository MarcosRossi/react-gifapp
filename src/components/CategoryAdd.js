import React, { useState } from 'react';
import PropTypes from 'prop-types';

const CategoryAdd = function ({ setCategory }) {
  const [categoryName, setcategoryName] = useState('');
  const handleChange = (e) => {
    setcategoryName(e.target.value);
  };
  const handleSubmit = (e) => {
    e.preventDefault();
    if (categoryName.trim().length <= 2) return;

    setCategory((cats) => [categoryName, ...cats]);
    setcategoryName('');
  };

  return (
    <form onSubmit={handleSubmit}>
      <input
        value={categoryName}
        onChange={handleChange}
        type='text'
        placeholder='Insert a new  category name'
      />
    </form>
  );
};

CategoryAdd.propTypes = {
  setCategory: PropTypes.func.isRequired,
};
export default CategoryAdd;
