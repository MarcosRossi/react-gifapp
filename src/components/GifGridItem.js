import React from 'react';
import PropTypes from 'prop-types';

export const GifGridItem = ({ title, imgs: url }) => {
  return (
    <div className='card animate__animated animate__fadeInDown'>
      <p>{title}</p>
      <img src={url} alt={title} />
    </div>
  );
};

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  imgs: PropTypes.string.isRequired,
};
