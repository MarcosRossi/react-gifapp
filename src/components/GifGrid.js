import React from 'react';
import { GifGridItem } from '../components/GifGridItem';
import { useFetchGifs } from '../hooks/useFetchGifs';
import PropTypes from 'prop-types';

export const GifGrid = ({ category = 'goku' }) => {
  const { data: images, loading } = useFetchGifs(category);

  return (
    <>
      <h3>{category}</h3>
      {loading && <p>Loading...</p>}
      <div className='card-grid'>
        {images.map((img) => (
          <GifGridItem {...img} key={img.id} />
        ))}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
