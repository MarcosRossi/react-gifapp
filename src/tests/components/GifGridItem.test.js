import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('test GifGridItem.js', () => {
  const props = {
    title: 'hunter x hunter GIF',
    imgs: 'https://test.com.ar',
  };
  const wrapper = shallow(<GifGridItem {...props} />);

  test('should create the component GifGridItem ', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should have a paragraph with a title', () => {
    const p = wrapper.find('p');
    
    expect(p.text().trim()).toBe(props.title);
  });

  test('should have a image with a src and alt props', () => {
    const img = wrapper.find('img');
    // props => Todas => img.props().src

    expect(img.prop('src')).toBe(props.imgs);
    expect(img.prop('alt')).toBe(props.title);
  });

  test('DIV should have the class animate__fadeInDown', () => {
    const div = wrapper.find('div');
    div.hasClass('animate__fadeInDown');

    expect(div.hasClass('animate__fadeInDown')).toBeTruthy();
  });
});
