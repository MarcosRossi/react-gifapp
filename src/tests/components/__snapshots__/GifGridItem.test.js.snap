// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test GifGridItem.js should create the component GifGridItem  1`] = `
<div
  className="card animate__animated animate__fadeInDown"
>
  <p>
    hunter x hunter GIF
  </p>
  <img
    alt="hunter x hunter GIF"
    src="https://test.com.ar"
  />
</div>
`;
